boefjes/boefjes/plugins/kat_rpki/main.py

```diff
@@ -25,9 +35,30 @@ def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
             exists = True
             expires = datetime.fromtimestamp(roa["expires"])
             roas.append({"prefix": roa["prefix"], "expires": expires.strftime("%Y-%m-%dT%H:%M"), "ta": roa["ta"]})
-            if expires > datetime.utcnow():
+            if expires > now:
                 valid = True
 
     results = {"vrps_records": roas, "valid": valid, "exists": exists}
 
     return [(set(), json.dumps(results))]
+
+
+def validate_age() -> Bool:
+    now = datetime.utcnow()
+    maxage = getenv("RPKI_CACHE_TIMEOUT") or RPKI_CACHE_TIMEOUT
+    with open(RPKI_META_PATH) as meta_file:
+        meta = json.load(meta_file)
+    cached_file_timestamp = datetime.strptime(meta["timestamp"], "%Y-%m-%dT%H:%M:%SZ")
+    return (now - cached_file_timestamp.total_seconds()) > maxage
+
+
+def refresh_rpki() -> Dict:
+    source_url = getenv("RPKI_SOURCE_URL") or RPKI_SOURCE_URL
+    response = requests.get(source_url, allow_redirects=True)
+    response.raise_for_status()
+    with open(RPKI_PATH, "w") as prki_file:
+        prki_file.write(response.content)
```

Comment:
There is a race condition that while opening/truncating and writing the file another boefje run might read an empty or partial file. Best way to avoid is to write to a temporary file and rename that file because renames are atomic.